pkg/cluster/snow.go

```diff
@@ -160,6 +193,31 @@ func getSnowMachineConfigs(ctx context.Context, client Client, c *Config) error
 		}
 
 		c.SnowMachineConfigs[machine.Name] = machine
+
+		if err := getSnowIPPools(ctx, client, c, machine); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func getSnowIPPools(ctx context.Context, client Client, c *Config, machine *anywherev1.SnowMachineConfig) error {
+	if c.SnowIPPools == nil {
+		c.SnowIPPools = map[string]*anywherev1.SnowIPPool{}
+	}
+
+	for _, dni := range machine.Spec.Network.DirectNetworkInterfaces {
+		if dni.IPPoolRef == nil {
+			continue
+		}
+
+		pool := &anywherev1.SnowIPPool{}
+		if err := client.Get(ctx, dni.IPPoolRef.Name, c.Cluster.Namespace, pool); err != nil {
+			return err
+		}
+
+		c.SnowIPPools[pool.Name] = pool
```

Comment:
you could check if this exists before making the client call to avoid extra `Get`'s when multiple machine configs use the same ip pool